module/Motion/CameraAction.cpp

```diff
@@ -28,25 +28,41 @@ void CameraAction::run()
              makeImageCommand, imageName);
   } else if(subject == CameraAction::Subject::PLARAIL) {
     sprintf(imageName, "Pla.jpeg");
-    sprintf(makeImageCommand, "plarail-image");
+    sprintf(makeImageCommand, "record");
     snprintf(cmd, 256, "cd etrobocon2024/front_camera && make %s && cd ../..", makeImageCommand);
   }
 
   system(cmd);
   printf("%s\n", cmd);
 
   // Fig_1の場合、物体検出を行うエンドポイントに画像をアップロードする
+  // それ以外の場合、画像をアップロードする処理をバックグラウンドで行う
+  // 実行コマンドの最後に「&」をつけることでバックグラウンドで実行する
+  // 参考：https://qiita.com/inosy22/items/341cfc589494b8211844
   char uploadImageName[20] = "Fig_1.jpeg";
-  if(strcmp(imageName, uploadImageName) == 0) {
+  if(subject == CameraAction::Subject::FIGURE && strcmp(imageName, uploadImageName) == 0) {
     snprintf(cmd, 256,
              "cd etrobocon2024 && make upload-detect FILE_PATH=front_camera/image_data/%s && cd ..",
              imageName);
     system(cmd);
     printf("%s\n", cmd);
-  } else {
+  } else if(subject == CameraAction::Subject::FIGURE) {
+    snprintf(
+        cmd, 256,
+        "(cd etrobocon2024 && make upload-image FILE_PATH=front_camera/image_data/%s && cd ..) &",
+        imageName);
+    system(cmd);
+    printf("%s\n", cmd);
+  }
+
+  // PLARAILが指定された場合には、画像の切り出し処理を行い無線通信デバイスに送信する処理をバックグラウンドで行う
+  if(subject == CameraAction::Subject::PLARAIL) {
+    sprintf(makeImageCommand, "plarail-image");
     snprintf(cmd, 256,
-             "cd etrobocon2024 && make upload-image FILE_PATH=front_camera/image_data/%s && cd ..",
-             imageName);
+             "(cd etrobocon2024/front_camera && make %s && cd ../.. &&"
+             "cd etrobocon2024 && make upload-image FILE_PATH=front_camera/image_data/%s && cd .., "
```

Comment:
同じところを何度もすみません。
ここ最後のworkspaceに戻るための"cd .."がカンマがついてしまって"cd ..,"となってしまっているのですが、これはworkspaceに戻れているやつですか？

```suggestion
             "cd etrobocon2024 && make upload-image FILE_PATH=front_camera/image_data/%s && cd .."
```

Reply:
すみません。,消し忘れです。特にエラーも出なかったので気づきませんでした。
ありがとうございます。